front/src/applications/operationalStudies/hooks/useSetupItineraryForTrainUpdate.ts

```diff
@@ -75,9 +78,66 @@ const useSetupItineraryForTrainUpdate = (
     osrdEditoastApi.endpoints.postInfraByInfraIdPathfindingBlocks.useLazyQuery();
   const [postPathProperties] =
     osrdEditoastApi.endpoints.postInfraByInfraIdPathProperties.useLazyQuery();
-  const { infraId } = useScenarioContext();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
 
   useEffect(() => {
+    const fetchPathStepsCoordinates = async (trainSchedule: TrainScheduleResult) => {
+      // get track sections
+      const trackSectionIds: string[] = [];
+      trainSchedule.path.forEach((step) => {
+        if ('track' in step) {
+          trackSectionIds.push(step.track);
+        }
+      });
+      const tracks = await getTrackSectionsByIds(trackSectionIds);
+
+      // get operational points
+      let ops: SearchResultItemOperationalPoint[] = [];
+      const searchPayload = buildOpSearchQuery(infraId, [trainSchedule]);
+
+      if (searchPayload) {
+        ops = (await postSearch({
+          searchPayload,
+          pageSize: 1000,
+        }).unwrap()) as SearchResultItemOperationalPoint[];
+      }
```

Comment:
We could also do that. Maybe it's safer if someone later somehow changes the value of `ops` because it's a `let`.

```suggestion
      const searchPayload = buildOpSearchQuery(infraId, [trainSchedule]);

      const ops = searchPayload
        ? ((await postSearch({
            searchPayload,
            pageSize: 1000,
          }).unwrap()) as SearchResultItemOperationalPoint[])
        : [];
```

Reply:
Done